### PR TITLE
Add R-like functions for fitting linear models

### PR DESCRIPTION
**Pull request by EpigeneMax**
## Description
This PR adds functions that behave like R `lm.fit` and `lm.wfit`.
They are merely wrappers around linear models fits from `statsmodels`, but provide quality-of-life improvement for users familiar with the R functions.

## Motivation and Context
This PR adds useful functions to later port `limma` fit functions.
See [the corresponding Jira card](https://epigenelabsteam.atlassian.net/browse/CR-263).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] No code (non-breaking change that does not impact code: formatting, build system, documentation...)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] My changes are in a separate branch named after a Jira card, or -- if not applicable -- with a descriptive name.
- [x] I have descriptive commit messages, with a short title (first line).
- [x] I have added and/or updated tests to reflect my changes.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have updated the [Changelog](../CHANGELOG.md).
- [x] My code and documentation follow the [numpy docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html).
- [x] I have reviewed possible licensing issues, preferably with a person in charge of the topic.

### CODE REVIEW

File: inmoose/utils/splines.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        boundary_knots = [np.min(x), np.max(x)]
+
+    if df is not None:
+        nIknots = df - 1 - include_intercept
+        if nIknots < 0:
+            nIknots = 0
+
+        if nIknots > 0:
+            knots = np.linspace(0, 1, num=nIknots + 2)[1:-1]
+            knots = np.quantile(x, knots)
```
Comment (guillaumeap) [inline inmoose/utils/splines.py:75]: What happends if knots is None and `nIknots > 0`?
It looks like knots is not redefined?

**Review by guillaumeap — COMMENTED** [review]
Looks good! I have a two questions :)
Also, do you want to create tests for splines.py?

File: inmoose/utils/splines.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        parameters do not depend on :code:`x`. Data can extend beyond
+        :code:`boundary_knots`.
+
+    Returns
+    -------
+    ndarray
+        a matrix of dimension :code:`(len(x), df)`, where :code:`df =
+        len(knots)-1-intercept` if :code:`df` was not supplied.
+    """
+    degree = 3
```
[the remark this answers is not in the document]
Comment (EpigeneMax) [reply inmoose/utils/splines.py]: we're building cubic splines, *i.e.* polynomials of degree 3.

File: inmoose/utils/splines.py (hunk 1 above)
Comment (EpigeneMax) [reply inmoose/utils/splines.py]: `knots` may remain `None`, in which case the spline is determined based on boundary conditions only.

File: inmoose/utils/splines.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        if include_intercept is False:
+            basis = basis[:, 1:]
+            const = const[:, 1:]
+
+        qr_const = np.linalg.qr(const.T, mode="complete")[0]
+        self.basis = (qr_const.T @ basis.T).T[:, 2:]
+
+    def predict(self, newx):
+        """
+        Evaluate the spline basis at given values
```
Comment (soleneweill) [inline inmoose/utils/splines.py:189]: Missing arg and return. Do you add return type usually?

**Review by soleneweill — APPROVED** [review]
Looks good, small comment on the documentation

File: inmoose/utils/splines.py (hunk 3 above)
Comment (EpigeneMax) [reply inmoose/utils/splines.py:189]: documentation updated

**Pull request merged**